flake.nix

```diff
@@ -0,0 +1,167 @@
+            "aarch64-unknown-linux-musl" 
+            "armv7-unknown-linux-musleabihf"
+          ];
+        };
+
+        craneLib = crane.lib.${system}.overrideToolchain rustToolchain;
+
+        # Common build inputs
+        nativeBuildInputs = with pkgs; [
+          # Rust toolchain and tools
+          rustToolchain
+          wasm-pack
+          
+          # Build tools
+          pkg-config
+          
+          # Compression and utilities
+          gzip
+          curl
+          wget
+          
+          # Development tools
+          git
+          jq
+          
+          # Cross-compilation support
+          gcc
+          
+          # For development convenience
+          cargo-watch
+          cargo-expand
+          bacon
+        ];
+
+        buildInputs = with pkgs; [
+          # System libraries that might be needed
+          openssl
+          sqlite
+        ] ++ lib.optionals stdenv.isDarwin [
+          # macOS specific dependencies
+          darwin.apple_sdk.frameworks.Security
+          darwin.apple_sdk.frameworks.SystemConfiguration
+        ];
+
+        # Environment variables
+        commonEnvVars = {
+          CARGO_TERM_COLOR = "always";
+          RUST_BACKTRACE = "1";
+          
+          # Cross-compilation environment
+          CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_LINKER = "${pkgs.pkgsStatic.stdenv.cc}/bin/cc";
+          CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_LINKER = "${pkgs.pkgsCross.aarch64-multiplatform.stdenv.cc}/bin/aarch64-unknown-linux-gnu-gcc";
+          CARGO_TARGET_ARMV7_UNKNOWN_LINUX_MUSLEABIHF_LINKER = "${pkgs.pkgsCross.armv7l-hf-multiplatform.stdenv.cc}/bin/arm-unknown-linux-gnueabihf-gcc";
+        };
+
+
+
+      in
+      {
+        # Development shells
+        devShells = {
+          default = pkgs.mkShell ({
+            inherit nativeBuildInputs buildInputs;
+            
+            shellHook = ''
+              echo "🔐 LLDAP Development Environment"
+              echo "==============================================="
+              echo "Rust version: ${rustVersion}"
+              echo "Standard cargo commands available:"
+              echo "  cargo build --workspace    - Build the workspace"
+              echo "  cargo test --workspace     - Run tests"
+              echo "  cargo clippy --tests --workspace -- -D warnings - Run linting"
+              echo "  cargo fmt --check --all    - Check formatting"
+              echo "  ./app/build.sh              - Build frontend WASM"
+              echo "  ./export_schema.sh          - Export GraphQL schema"
+              echo "==============================================="
+              echo ""
+              
+              # Ensure wasm-pack is available
+              if ! command -v wasm-pack &> /dev/null; then
+                echo "⚠️  wasm-pack not found in PATH"
+              fi
+              
+              # Check if we're in the right directory
+              if [[ ! -f "Cargo.toml" ]]; then
```

Comment:
I think we should use `git rev-parse --show-toplevel` instead, to get the top-level directory. A lot of the sub-directories contain Cargo.toml as well.